test-auth.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+
+TAG="$1"
+IMG="$2"
+
+case "$TAG" in
+  4.1 )
+    ES_VERSION=1.5
+    ;;
+  4.4 )
+    ES_VERSION=2.2
+    ;;
+  * )
+    ES_VERSION="$TAG"
+    ;;
+esac
+
+DB_CONTAINER="kbelastic"
+DATA_CONTAINER="${DB_CONTAINER}-data"
+ES_IMG="quay.io/aptible/elasticsearch:${ES_VERSION}"
+KIBANA_CONTAINER="kibana-${TAG}"
+KIBANA_CREDS=kuser:kpass
+ES_USER=euser 
+ES_PASS=epass
+
+function cleanup {
+  echo "Cleaning up"
+  docker rm -f "$DB_CONTAINER" "$DATA_CONTAINER" "$KIBANA_CONTAINER" > /dev/null 2>&1 || true
+}
+
+function wait_for_reqeust {
+  CONTAINER=$1
+  REQUEST=$2
+  for _ in $(seq 1 30); do
+    if docker exec -it "$CONTAINER" curl -fs "$REQUEST" > /dev/null 2>&1; then
+      return 0
+    fi
+    sleep 1
+  done
+
+  echo "No response"
+  docker logs "$CONTAINER"
+  return 1
+}
+
+trap cleanup EXIT
+cleanup
+
+echo "Creating data container"
+docker create --name "$DATA_CONTAINER" "${ES_IMG}"
+
+echo "Starting DB"
+docker run -it --rm \
+  -e USERNAME=${ES_USER} -e PASSPHRASE=${ES_PASS} -e DATABASE=db \
+  --volumes-from "$DATA_CONTAINER" \
+  "${ES_IMG}" --initialize \
+  >/dev/null 2>&1
+
+docker run -d --name="$DB_CONTAINER" \
+  --volumes-from "$DATA_CONTAINER" \
+  "${ES_IMG}"
+
+echo "Waiting for DB to come online"
+wait_for_reqeust "$DB_CONTAINER" "http://localhost:9200"
+
+ES_IP=$(docker inspect ${DB_CONTAINER} | grep '\"IPAddress\"' | head -n 1 | awk -F "\"" '{print $4}')
+
+echo "Verifying DB shutdown message isn't present"
+docker logs "$DB_CONTAINER" 2>&1 | grep -vqiE "node.+closed"
+
+echo "Sarting Kibana"
+docker run -d --name="$KIBANA_CONTAINER" \
+  -e DATABASE_URL="http://${ES_USER}:${ES_PASS}@${ES_IP}:80" \
+  -e AUTH_CREDENTIALS="$KIBANA_CREDS" \
+  -e TESTING="true" \
+  -p 5601:5601 \
```

Comment:
Not a fan of using a specific port here - consider `docker exec`'ing into the Kibana container